kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/adapter/QuickConversationAdapter.java

```diff
@@ -250,6 +251,14 @@ public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
                     showConversationSourceIcon(channel, myholder.attachmentIcon);
                 }
 
+                if(TextUtils.isEmpty(myholder.messageTextView.getText()))
+                {
```

Comment:
This check is not correct, it might fail for some slow Android devices, during repeated scrolling.
Rather than checking if messageTextView is empty (which we set empty in previous lines). Add a else if statement which checks if the message is a Rich message.
Message.isRichMessage() should tell you that.